lib/Source.ml

```diff
@@ -231,15 +231,15 @@ let string_literal t mode (l : Location.t) =
   let toks =
     tokens_at t
       ~filter:(function
-        | Parser.STRING (_, None) -> true
+        | Parser.STRING (_, _, None) -> true
         | Parser.LBRACKETAT | Parser.LBRACKETATAT | Parser.LBRACKETATATAT ->
             true
         | _ -> false )
       l
   in
   match toks with
-  | [(Parser.STRING (_, None), loc)]
-   |(Parser.STRING (_, None), loc)
+  | [(Parser.STRING (_, _, None), loc)]
+   |(Parser.STRING (_, _, None), loc)
```

Comment:
It seems like these datatypes are taken from `compiler-libs` and not from `ocaml-migrate-parsetree`. Is there any chance we could use the parsetree instead?

Otherwise, would it make sense to simply allow `Parser.STRING _`? (i.e. `"str"`, `{|str|}`, `{delim|str|delim}`, … and not just `"str"`)

Reply:
That's a good point. OMP does not embed the different parsers, and when we reparse bits of the input like here, it's always using the parser from compiler libs. We've talked in the past of packaging a fixed recent version of the parser instead of migrating the AST from the compiler-libs version, that's another point in favor of this.

Reply:
as discussed with @kit-ty-kate, this function is used to determine the string's precise loc, which is now available in the 4.11 AST, so it might be possible to get rid of most of it!

Reply:
I opened #1392 for this part